### ISSUE TITLE
Authorisation Server

### ISSUE THREAD

**Issue by th0mas** [dwyl/smart-home-security-system#3]
#1 References a device authorisation server to verify each device that accesses the door.

We need to establish some objectives for this server to achieve.

@nelsonic is this going to integrate with the existing Dwyl [auth](https://github.com/dwyl/auth) system? 

I think the authorisation server should implement the following features

- [ ] Establish a unique device ID for each device
- [ ] Establish a shared "secret" with each device -e.g. to authenticate the device after a push notification.
- [ ] Be able to link each "user" with multiple devices
- [ ] Log *all* authorisation attempts
- [ ] Verify a device in a "low security event" (internal doors)
- [ ] Send a push notification in a "high security event" (external doors)
- [ ] Verify device after 2FA has been completed

Anything that needs to be added/removed from this? Otherwise I'll start work on building at least a skeleton of this now so we can start to integrate the Raspberry Pi's with it later.

**Comment by nelsonic** [comment]
@th0mas yeah, our longer term vision for the @home (Native Mobile) App is that it contains additional functionality that is useful to the people visiting/living in the co-living house. So yes, using the dwyl auth system would be ideal from day 1.
Very happy to remote-pair with you on anything that needs to be built, equally happy to let you get on with it.
It would be insightful to get your perspective on the dwyl auth system as a fresh pair of eyes. 👍 

The list of features/acceptance criteria you have listed seems great as a starting point.
Please open more issues as you go e.g. if you get stuck or have questions.
